### PR TITLE
CI - Use new action axelerant/platformsh-action

### PR DESCRIPTION
**Pull request by zeshanziya**
1. use new action axelerant/platformsh-action for deployment
2. Add workflow for cleaning platform.sh env on PR close

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -165,8 +165,14 @@ jobs:
     # Dependabot PR's can't access secrets, so we can't deploy.
     if: github.actor != 'dependabot[bot]'
     steps:
-      - uses: axelerant/platformsh-deploy-action@main
+      - name: Check out repository code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Deploy to platform.sh
+        uses: axelerant/platformsh-action@main
```
Comment (hussainweb) [inline .github/workflows/ci.yml:173]: Let's pin to `v1`?

Comment (zeshanziya) [reply .github/workflows/ci.yml]: done.

File: .github/workflows/pr-close.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  pull_request:
+    types: [closed]
+
+jobs:
+  on-pr-close:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Clean Platform.sh env if available
+      uses: axelerant/platformsh-action@main
```
Comment (hussainweb) [inline .github/workflows/pr-close.yml:13]: @zeshanziya, missed this.

Comment (hussainweb) [reply .github/workflows/pr-close.yml:13]:
```suggestion
      uses: axelerant/platformsh-action@v1
```

**Pull request merged**